src/xref2/component.mli

```diff
@@ -478,7 +478,11 @@ module Element : sig
   type exception_ = [ `Exception of Identifier.Exception.t * Exception.t ]
 
   type extension =
-    [ `Extension of Identifier.Extension.t * Extension.Constructor.t ]
+    [ `Extension of
+      Identifier.Extension.t * Extension.Constructor.t * Extension.t ]
+
+  type extension_decl =
+    [ `ExtensionDecl of Identifier.Extension.t * Extension.Constructor.t ]
```

Comment:
I know its not the case in the rest of the codebase, but I think these require comments to explain what each field corresponds to. There is such a comment in paths.mli, which is very nice.